src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder.cs

```diff
@@ -70,6 +70,20 @@ internal override BoundStatement BindSwitchExpressionAndSections(SwitchStatement
                 locals, functions, switchSections, defaultLabel, this.BreakLabel, this);
         }
 
+        internal override void BindPatternSwitchLabelForInference(CasePatternSwitchLabelSyntax node, DiagnosticBag diagnostics)
+        {
+            // This simulates enough of the normal binding path of a switch statement to cause
+            // the label's pattern variables to have their types inferred, if necessary.
+            BoundPatternSwitchLabel defaultLabel = null;
+            BindPatternSwitchSectionLabel(
```

Comment:
>BindPatternSwitchSectionLabel [](start = 12, length = 29)

I think we should make sure that the ```node``` actually belongs to the switch statement owned by this binder.